### DISCUSSION TITLE
What is the best way to work with a live changing document?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#69469] [Help] [answered]
### Summary

Ok... once upon a time I used MongoDB, but I had problems connecting to Next.js, and I ended up using json documents from the same site.   These Json documents don't change, so the data stays as it and it works fine without using a 3rd party data base.

Now, what if I need to make changes / adding information to the document?  This document is also not meant to be shared in the front end.

^^ Would I need to use something like MongoDB if I need to make changes????

Can I use the File System???  

What do I want to do exactly????

When a user visits a dynamic link, I want to add a record that this dynamic link has been visit, and save the information in the document.

^^  Now, what if I want to see the data later on?  How do I access it from the live site if I am using the File System?

Do I need to create my own API if I don't want to use MongoDB???

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Using file system, is probably not a good idea. It is better to use a database. 

There are lots of databases available, for example: https://vercel.com/docs/storage/vercel-kv 

MongoDB is often famous because it is taught in bootcamps and such, but it is really up to you to decide which solution works best for you.

**Reply by Elindo586** [reply]
I am likely gonna use Postgresql connected to Vercel.

Just curious. Why do you say using the file system might not be a good idea?..

**Reply by icyJoseph** [reply]
Well, you'd end up recreating a mini database API, that interacts with the file system, and that shouldn't be your concern really. You should focus on your business logic. Unless you have the time, resources and motivation to do so.

Like, everything should be taken with a grain of salt, it is not intrinsically bad, nor totally wrong. However, accessing a file handle, is somewhat slow. You'd also need to make sure that the transactions you make with the file system are atomic (imagine two views arrive, at the same time, you'd need to make sure that you increment your view counter, properly). Also, if you need to work with the production data, and with a database, that'd be as simple as changing the connection, whereas a file system... well, you'd need to somehow gain access to that disk. Concurrency is tough.

Also, your data, on disk, is now accessible, by other parties, a process, might delete the database file, or change its name, who knows...

Also, if you are deploying to Vercel, your deploy is serverless, so the files included are destroyed, and not guaranteed to persist. 

- https://github.com/orgs/vercel/discussions/241#discussioncomment-363002
